File: examples/CSharp/RealtimeReporting/InfluxDbReporting.cs
using System;
using System.Threading.Tasks;

using NBomber.Contracts;
using NBomber.CSharp;
using NBomber.Sinks.InfluxDB;

namespace CSharp.RealtimeReporting
{
    public class InfluxDbReporting
    {
        public static void Run()
        {
            var step = Step.Create("step", async context =>
            {
                await Task.Delay(TimeSpan.FromSeconds(0.1));
                return Response.Ok(sizeBytes: 100);
            });

            var scenario = ScenarioBuilder
                .CreateScenario("simple_scenario", step)
                .WithoutWarmUp()
                .WithLoadSimulations(Simulation.KeepConstant(1, TimeSpan.FromMinutes(1)));

            var influxDb = new InfluxDBSink("http://localhost:8086", "default");

            NBomberRunner
                .RegisterScenarios(scenario)
                .WithTestSuite("reporting")
                .WithTestName("influx_test")
                .WithReportingSinks(
                    reportingSinks: new[] {influxDb},
                    sendStatsInterval: TimeSpan.FromSeconds(10)
                )
                .Run();
        }
    }
}

File: examples/FSharpProd/RealtimeReporting/InfluxDbReporting.fs
module FSharpProd.RealtimeReporting.InfluxDbReporting

open System.Threading.Tasks
open FSharp.Control.Tasks.NonAffine
open NBomber
open NBomber.Contracts
open NBomber.FSharp
open NBomber.Sinks.InfluxDB

let run () =

    let step = Step.create("step_1", fun context -> task {

        do! Task.Delay(seconds 1)

        // this message will be saved to elastic search
        context.Logger.Debug("hello from NBomber")

        return Response.ok()
    })

    let influxConfig = InfluxDbSinkConfig.create(url = "http://localhost:8086", database = "default")
    use influxDb = new InfluxDBSink(influxConfig)

    Scenario.create "hello_world_scenario" [step]
    |> Scenario.withoutWarmUp
    |> Scenario.withLoadSimulations [KeepConstant(copies = 1, during = minutes 1)]
    |> NBomberRunner.registerScenario
    |> NBomberRunner.withTestSuite "reporting"
    |> NBomberRunner.withTestName "influx_test"
    |> NBomberRunner.withReportingSinks [influxDb] //TODO???:  (seconds 10)
    |> NBomberRunner.run
    |> ignore
